src/Socket_p.h

```diff
@@ -352,7 +352,7 @@ namespace Arcus
 
         {
             std::lock_guard<std::mutex> lk(receiveQueueMutexBlock);
-            message_received_condition_variable = true;
+            message_received_condition_variable.notify_all();
```

Comment:
Is there a reason this is 'notify_all' instead of 'notify_one'? (But perhaps it's safer to keep it like this if it works?)